### PR TITLE
Add File Doctype and Corresponding Components to handle them

### PR DESCRIPTION
**Pull request by nurdtechie98**
The following functionalities have been implemented 
- [x] uploading the files to backend 
- [x] display the filed using listandform component
- [x] delete functionality
- [ ] redundancy removal in files by hashing 
- [ ] ui improvement

review it and let me know for changes @netchampfaris @surajshetty3416

### CODE REVIEW

File: server/file.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const frappe = require("frappejs")
+var formidable = require("formidable")
```
Comment (surajshetty3416) [inline server/file.js]: Please use either `const` or `let` (to avoid inconsistent codebase)

Use `let` if the variable needs to be reassigned else `const`

File: server/file.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        frappe.insert({
+            doctype:'FileContent',
+            name:fields.image_title,
+            path:"accounting/static/"+files.filetoupload.name
+            })
+        fs.rename(oldpath, newpath, function(err){
+            if (err) throw err;
+        });
+    });
+    return "File Upload Successfulls";
```
Comment (surajshetty3416) [inline server/file.js]: ?

File: server/index.js [hunk 3]
```diff
@@ -16,6 +16,7 @@ const fs = require('fs');
 const { setupExpressRoute: setRouteForPDF } = require('frappejs/server/pdf');
 const auth = require('./../auth/auth')();
 const morgan = require('morgan')
+const posRegister = require('./file');
```
Comment (surajshetty3416) [inline server/index.js]: file is not related to pos .. 

you can rename it to `setupFileMethods`

File: ui/components/List/List.vue [hunk 4]
```diff
@@ -29,7 +29,7 @@ import ListItem from './ListItem';
 
 export default {
   name: 'List',
-  props: ['doctype'],
+  props: ['doctype','newDoc'],
```
Comment (surajshetty3416) [inline ui/components/List/List.vue]: why `newDoc` in props ?

File: ui/components/List/List.vue [hunk 5]
```diff
@@ -87,13 +81,14 @@ export default {
...
+    reqOpenForm(name) {
       this.activeItem = name;
-      this.$router.push(`/edit/${this.doctype}/${name}`);
+      this.$emit("openForm",name);
     },
     async deleteCheckedItems() {
-      await frappe.db.deleteMany(this.doctype, this.checkList);
+      await frappe.db.deleteMany(this.doctype,this.checkList);
       this.checkList = [];
+      this.$emit('deleted',this.checkList);
```
Comment (surajshetty3416) [inline ui/components/List/List.vue]: Move this above `this.checkList = []`.  Checklist should be cleared after emitting the event

File: ui/pages/File/FListAndForm.vue [hunk 6]
```diff
@@ -0,0 +1,77 @@
+<template>
+    <div class="frappe-list-form row no-gutters">
+        <div class="col-4 border-right">
+            <frappe-list :doctype="doctype" :filters="filters" :newDoc="openNewDoc" :key="doctype" @openForm="onOpenForm" />
```
Comment (surajshetty3416) [inline ui/pages/File/FListAndForm.vue]: :newDoc should be replaced with some listener

File: ui/routes/index.js [hunk 7]
```diff
@@ -1,4 +1,6 @@
 import ListAndForm from '../pages/ListAndForm';
+import Fpick from '../pages/File/Fpick';
+import FListAndForm from '../pages/File/FListAndForm'
```
Comment (surajshetty3416) [inline ui/routes/index.js]: Suggestion:  Set bit descriptive names
like 
Fpick -> FilePicker
FListAndForm -> FileManager or just Files

**Comment by surajshetty3416** [comment]
Please also add animated gif of working feature

File: ui/components/List/List.vue (hunk 4 above)
Comment (nurdtechie98) [reply ui/components/List/List.vue]: Ohkk will add a listener here instead

File: server/index.js (hunk 3 above)
Comment (nurdtechie98) [reply server/index.js]: oops!,these are the naming before I moved to frappes earlier I had them in accounting/pos.

**Comment by nurdtechie98** [comment]
filetest

**Comment by nurdtechie98** [comment]
filetest
@surajshetty3416 thanks for the review I have made the required changes, do let me know for further changes
